drivers/spi/spi_ll_stm32.c

```diff
@@ -867,6 +984,11 @@ static int transceive_dma(const struct device *dev,
 	}
 
 #if DT_HAS_COMPAT_STATUS_OKAY(st_stm32h7_spi)
+	/* Set TSIZE = 0 just in case, somehow, it has a value != 0 from the previous transaction
+	 * (which is a non-tested scenarion for DMA).
+	 */
+	LL_SPI_SetTransferSize(spi, 0);
```

Comment:
Is it safe operation to perform when fifo is not used ?
Why not doing it under fifo-enabled condition ?

Reply:
The purpose of this was to make the SPI use safer, let me explain. DMA, as it stands now, has only been used in 'endless' mode, that is, with TSIZE = 0. If a 'normal' SPI operation occurs, and after a DMA one is performed, maybe due to some error or malfunction TSIZE could end up with a value different than 0 at the beginning of the DMA transaction. In that case, it would be safer to set TSIZE to 0, as that's the only scenario in which DMA has been tested.

It also adds the semantics to the code of that DMA operations have only been tested under this particular scenario.

As it stands now, it's not possible to perform a non-DMA SPI op. and then a DMA one, because DMA is enabled in Kconfig and then all SPI transactions are necessarily DMA. In that sense, the function call above is unnecessary, yet I added it in case this changes in the future and to add semantics to the code as explained before.